src/render.c

```diff
@@ -1466,19 +1503,22 @@ void free_root_tile(session_t *ps) {
 
 void deinit_render(session_t *ps) {
 	// Free alpha_picts
-	for (int i = 0; i <= MAX_ALPHA; ++i)
-		free_picture(ps->c, &ps->alpha_picts[i]);
+	for (int i = 0; i <= MAX_ALPHA; ++i) {
+		x_free_picture(&ps->c, ps->alpha_picts[i]);
+	}
 	free(ps->alpha_picts);
 	ps->alpha_picts = NULL;
 
 	// Free cshadow_picture and black_picture
-	if (ps->cshadow_picture == ps->black_picture)
+	if (ps->cshadow_picture == ps->black_picture) {
```

Comment:
silly github doesn't let me add a suggestion to proper lines
```suggestion
	if (ps->cshadow_picture != ps->black_picture) {
		x_free_picture(&ps->c, ps->cshadow_picture);
	}
	
	x_free_picture(&ps->c, ps->black_picture);
	x_free_picture(&ps->c, ps->white_picture);
	ps->cshadow_picture = ps->black_picture = ps->white_picture = XCB_NONE;
```